gh-134733: Fix documentation for the show_empty option of ast.dump() (#134925)
python/cpython: Doc/library/ast.rst

   indents that many spaces per level.  If *indent* is a string (such as ``"\t"``),
   that string is used to indent each level.

>    Unless *show_empty* is true, optional empty lists will be

Reviewer: Since `Unlcess *show_empty* is true...` is the same as `If *show_empty if False (the default)`, can't we just merge these seperate statements?

Reviewer 2: Yes, I think the old wording is clearer here:
```suggestion
   If *show_empty* is false (the default), optional empty lists will be
```
